qiskit_experiments/data_processing/nodes.py

```diff
@@ -0,0 +1,173 @@
+        """
+        Args:
+            scale: Float with which to multiply the IQ data.
+            validate: If set to False the DataAction will not validate its input.
+        """
+        self.scale = scale
+        super().__init__(validate)
+
+    @abstractmethod
+    def _process(self, datum: np.array) -> np.array:
+        """Defines how the IQ point will be processed.
+
+        Args:
+            datum: A 3D array of shots, qubits, and a complex IQ point as [real, imaginary].
+
+        Returns:
+            Processed IQ point.
+        """
+
+    def _format_data(self, datum: Any, validate: bool = True) -> Any:
+        """Check that the IQ data has the correct format and convert to numpy array.
+
+        Args:
+            datum: A single item of data which corresponds to single-shot IQ data. It should
+                have dimension three: shots, qubits, iq-point as [real, imaginary].
+            validate: If True the DataAction checks that the format of the datum is valid.
+
+        Returns:
+            datum as a numpy array.
+
+        Raises:
+            DataProcessorError: If the datum does not have the correct format.
+        """
+        datum = np.asarray(datum, dtype=float)
+
+        if validate and len(datum.shape) != 3:
+            raise DataProcessorError(
+                f"Single-shot data given {self.__class__.__name__}"
+                f"must be a 3D array. Instead, a {len(datum.shape)}D "
+                f"array was given."
+            )
+
+        return datum
+
+    def __repr__(self):
+        """String representation of the node."""
+        return f"{self.__class__.__name__}(validate: {self._validate}, scale: {self.scale})"
+
+
+class ToReal(IQPart):
+    """IQ data post-processing. Isolate the real part of the IQ data."""
+
+    def _process(self, datum: np.array) -> np.array:
+        """Take the real part of the IQ data.
+
+        Args:
+            datum: A 3D array of shots, qubits, and a complex IQ point as [real, imaginary].
+
+        Returns:
+            A 2D array of shots, qubits. Each entry is the real part of the given IQ data.
+        """
+        if self.scale is None:
+            return datum[:, :, 0]
+
+        return datum[:, :, 0] * self.scale
+
+
+class ToImag(IQPart):
+    """IQ data post-processing. Isolate the imaginary part of the IQ data."""
+
+    def _process(self, datum: np.array) -> np.array:
+        """Take the imaginary part of the IQ data.
+
+        Args:
+            datum: A 3D array of shots, qubits, and a complex IQ point as [real, imaginary].
+
+        Returns:
+            A 2D array of shots, qubits. Each entry is the imaginary part of the given IQ data.
+        """
+        if self.scale is None:
+            return datum[:, :, 1]
+
+        return datum[:, :, 1] * self.scale
+
+
+class Probability(DataAction):
+    """Count data post processing. This returns qubit 1 state probabilities."""
```

Comment:
What do you mean by "qubit 1 state probabilities" ?

Reply:
Changed here: 4c9acae